### PR TITLE
improve 4xx voluptuous message

### PR DESCRIPTION
**Pull request by tanyunshi**
/!\ WIP ONLY

Related to https://github.com/gnocchixyz/gnocchi/issues/195

### CODE REVIEW

**Comment by tanyunshi** [comment]
hop ! o/ tests passed !

File: gnocchi/tests/functional/gabbits/aggregates-with-metric-ids.yaml [hunk 1]
```diff
@@ -774,8 +780,11 @@ tests:
       data:
         operations: "(rolling blah 2 (metric $HISTORY['create metric1'].$RESPONSE['$.id'] mean))"
       status: 400
-      response_strings:
-        - "Invalid input: 'rolling' operation invalid for dictionary value"
+      response_json_paths:
+        $.description.cause: "Invalid input"
+        # Use regex to avoid the 'u' populated by voluptuous
```
Comment (chungg) [inline gnocchi/tests/functional/gabbits/aggregates-with-metric-ids.yaml]: not entire sure what this comment is about? something about unicode? i would remove the copy/paste but not a big deal.

File: gnocchi/tests/functional/gabbits/resource-type.yaml [hunk 2]
```diff
@@ -54,12 +55,8 @@ tests:
...
                   noexist: foo
       status: 400
-      response_strings:
-          # NOTE(sileht): We would prefer to have a better message but voluptuous seems a bit lost when
-          # an Any have many dict with the same key, here "type"
-          # - "Invalid input: extra keys not allowed @ data[u'attributes'][u'foo'][u'noexist']"
-          # - "Invalid input: not a valid value for dictionary value @ data[u'attributes'][u'foo'][u'type']"
-          - "Invalid input:"
+      response_json_paths:
+        $.description.cause: "Invalid input"
```
Comment (chungg) [inline gnocchi/tests/functional/gabbits/resource-type.yaml]: does this still not include detail/reason?

Comment (tanyunshi) [reply gnocchi/tests/functional/gabbits/resource-type.yaml]: :+1:  Actually sileht's comment is still valid.

File: gnocchi/tests/functional/gabbits/aggregates-with-metric-ids.yaml (hunk 1 above)
Comment (tanyunshi) [reply gnocchi/tests/functional/gabbits/aggregates-with-metric-ids.yaml]: You are right, it's about the unicode: the sentences contain 'u' in python2. :roll_eyes: 

I removed the comments and mention this in a commit message. ^^

**Pull request merged**